### PR TITLE
feat: show create account dialog if no GodWoken account

### PR DESCRIPTION
**Pull request by wobsoriano**
For now, we added the error to the `usePoolData` hook since it's able to catch the JSON RPC error. It's not that invasive since we are checking the `chainId` too before showing the modal:

```ts
const [, , jsonRpcError] = usePoolData(D4_POOL_NAME)
const { chainId } = useActiveWeb3React()

const hasNoAccount = useMemo(() => {
  if (!jsonRpcError) return false
  return (
    (jsonRpcError as SerializedEthereumRpcError)?.error?.code === -32603 &&
    chainId === 71401
  )
}, [jsonRpcError, chainId])
```

### CODE REVIEW

File: src/components/NoAccountModal.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+export default function NoAccountModal({ open }: Props): ReactElement {
+  const [, , jsonRpcError] = usePoolData(D4_POOL_NAME)
+  const { chainId } = useActiveWeb3React()
+
+  const hasNoAccount = useMemo(() => {
+    if (!jsonRpcError) return false
+    return (
+      (jsonRpcError as SerializedEthereumRpcError)?.error?.code === -32603 &&
+      chainId === 71401
```
Comment (dezzyboy) [inline src/components/NoAccountModal.tsx:59]: may be a better idea to  utilize ChainId variable from constants rather than hardcoding it

**Review by dezzyboy — APPROVED** [review]
other than the single suggestion this looks good to merge

File: src/components/NoAccountModal.tsx (hunk 1 above)
Comment (wobsoriano) [reply src/components/NoAccountModal.tsx]: Done!

**Review by dezzyboy — APPROVED** [review]

File: src/components/NoAccountModal.tsx (hunk 1 above)
Comment (dezzyboy) [reply src/components/NoAccountModal.tsx]: great looks good thanks 👍🏾

**Pull request merged**